File: kt/task/03boolean.kt
package task

import kotlin.math.abs

class BooleanTask {

    var datScanner = scanner.Scanner()
    var ansScanner = scanner.Scanner()

    fun solve(pathPrefix: String) {
        for (taskNo in 1..40) {
            var taskNoStr = "%03d".format(taskNo)
            var fileName: String
            for (testNo in 1..100) {
                var testNoStr = "%03d".format(testNo)
                fileName = "$pathPrefix/03boolean/Boolean$taskNoStr/$testNoStr/$testNoStr"
                if (!datScanner.changeReader("$fileName.dat")) {
                    println("Boolean$taskNoStr couldn't open datScanner source")
                    return
                }
                if (!ansScanner.changeReader("$fileName.ans")) {
                    println("Boolean$taskNoStr couldn't open ansScanner source")
                    return
                }
                val ok = when (taskNo) {
                    1 -> this.boolean1()
                    2 -> this.boolean2()
                    3 -> this.boolean3()
                    4 -> this.boolean4()
                    5 -> this.boolean5()
                    6 -> this.boolean6()
                    7 -> this.boolean7()
                    8 -> this.boolean8()
                    9 -> this.boolean9()
                    10 -> this.boolean10()
                    11 -> this.boolean11()
                    12 -> this.boolean12()
                    13 -> this.boolean13()
                    14 -> this.boolean14()
                    15 -> this.boolean15()
                    16 -> this.boolean16()
                    17 -> this.boolean17()
                    18 -> this.boolean18()
                    19 -> this.boolean19()
                    20 -> this.boolean20()
                    21 -> this.boolean21()
                    22 -> this.boolean22()
                    23 -> this.boolean23()
                    24 -> this.boolean24()
                    25 -> this.boolean25()
                    26 -> this.boolean26()
                    27 -> this.boolean27()
                    28 -> this.boolean28()
                    29 -> this.boolean29()
                    30 -> this.boolean30()
                    31 -> this.boolean31()
                    32 -> this.boolean32()
                    33 -> this.boolean33()
                    34 -> this.boolean34()
                    35 -> this.boolean35()
                    36 -> this.boolean36()
                    37 -> this.boolean37()
                    38 -> this.boolean38()
                    39 -> this.boolean39()
                    40 -> this.boolean40()
                    else -> false
                }
                scanner.removeFiles()
                if (!ok) {
                    println("Boolean$taskNoStr Test#$testNoStr has not been tested")
                    return
                }
                if (!datScanner.eof()) {
                    println("Boolean$taskNoStr Test#$testNoStr input is not enough")
                    return
                }
                if (!ansScanner.eof()) {
                    println("Boolean$taskNoStr Test#$testNoStr output is not enough")
                    return
                }
            }
            println("Boolean$taskNoStr has been tested!")
        }
    }

    private fun boolean1(): Boolean {
        val a = this.datScanner.nextInt()
        val isPositive = a > 0
        return this.ansScanner.checkBoolean(isPositive)
    }

    private fun boolean2(): Boolean {
        val a = this.datScanner.nextInt()
        val isOdd = a % 2 != 0
        return this.ansScanner.checkBoolean(isOdd)
    }

    private fun boolean3(): Boolean {
        val a = this.datScanner.nextInt()
        val isEven = a % 2 == 0
        return this.ansScanner.checkBoolean(isEven)
    }

    private fun boolean4(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val result = a > 2 && b <= 3
        return this.ansScanner.checkBoolean(result)
    }

    private fun boolean5(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val result = a >= 0 || b < -2
        return this.ansScanner.checkBoolean(result)
    }

    private fun boolean6(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val c = this.datScanner.nextInt()
        val result = b in (a + 1) until c
        return this.ansScanner.checkBoolean(result)
    }

    private fun boolean7(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val c = this.datScanner.nextInt()
        val result = b in (a + 1) until  c || b in (c + 1) until a
        return this.ansScanner.checkBoolean(result)
    }

    private fun boolean8(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val eachOdd = a % 2 != 0 && b % 2 != 0
        return this.ansScanner.checkBoolean(eachOdd)
    }

    private fun boolean9(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val anyOdd = a % 2 != 0 || b % 2 != 0
        return this.ansScanner.checkBoolean(anyOdd)
    }

    private fun boolean10(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val oneOdd = (a + b) % 2 != 0
        return this.ansScanner.checkBoolean(oneOdd)
    }

    private fun boolean11(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val equalParity = (a + b) % 2 == 0
        return this.ansScanner.checkBoolean(equalParity)
    }

    private fun boolean12(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val c = this.datScanner.nextInt()
        val eachPositive = a > 0 && b > 0 && c > 0
        return this.ansScanner.checkBoolean(eachPositive)
    }

    private fun boolean13(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val c = this.datScanner.nextInt()
        val anyPositive = a > 0 || b > 0 || c > 0
        return this.ansScanner.checkBoolean(anyPositive)
    }

    private fun boolean14(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val c = this.datScanner.nextInt()
        val onePositive = (a > 0 && b <= 0 && c <= 0) ||
                (b > 0 && a <= 0 && c <= 0) ||
                (c > 0 && a <= 0 && b <= 0)
        return this.ansScanner.checkBoolean(onePositive)
    }

    private fun boolean15(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val c = this.datScanner.nextInt()
        val twoPositive = (a > 0 && b > 0 && c <= 0) ||
                (b > 0 && c > 0 && a <= 0) ||
                (c > 0 && a > 0 && b <= 0)
        return this.ansScanner.checkBoolean(twoPositive)
    }

    private fun boolean16(): Boolean {
        val number = this.datScanner.nextInt()
        val isTwoDigitalEven = number in 10..99 && number % 2 == 0
        return this.ansScanner.checkBoolean(isTwoDigitalEven)
    }

    private fun boolean17(): Boolean {
        val number = this.datScanner.nextInt()
        val threeDigitalOdd = number in 100..999 && number % 2 != 0
        return this.ansScanner.checkBoolean(threeDigitalOdd)
    }

    private fun boolean18(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val c = this.datScanner.nextInt()
        val hasPair = a == b || b == c || c == a
        return this.ansScanner.checkBoolean(hasPair)
    }

    private fun boolean19(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val c = this.datScanner.nextInt()
        val hasOppositePair = a == -b || b == -c || c == -a
        return this.ansScanner.checkBoolean(hasOppositePair)
    }

    private fun boolean20(): Boolean {
        val number = this.datScanner.nextInt()
        val hundreds = number / 100
        val tens = number / 10 % 10
        val ones = number % 10
        val areDifferent = hundreds != tens && tens != ones && ones != hundreds
        return this.ansScanner.checkBoolean(areDifferent)
    }

    private fun boolean21(): Boolean {
        val number = this.datScanner.nextInt()
        val hundreds = number / 100
        val tens = number / 10 % 10
        val ones = number % 10
        val areAscending = tens in (hundreds + 1) until ones
        return this.ansScanner.checkBoolean(areAscending)
    }

    private fun boolean22(): Boolean {
        val number = this.datScanner.nextInt()
        val hundreds = number / 100
        val tens = number / 10 % 10
        val ones = number % 10
        val areOrdered = tens in (hundreds + 1) until ones ||
                tens in (ones + 1) until hundreds
        return this.ansScanner.checkBoolean(areOrdered)
    }

    private fun boolean23(): Boolean {
        val number = this.datScanner.nextInt()
        val thousands = number / 1000
        val hundreds = number / 100 % 10
        val isPalindrome = hundreds * 10 + thousands == number % 100
        return this.ansScanner.checkBoolean(isPalindrome)
    }

    private fun boolean24(): Boolean {
        val a = this.datScanner.nextDouble()
        val b = this.datScanner.nextDouble()
        val c = this.datScanner.nextDouble()
        val d = b * b - 4 * a * c
        val hasRoot = d >= 0
        return this.ansScanner.checkBoolean(hasRoot)
    }

    private fun boolean25(): Boolean {
        val x = this.datScanner.nextDouble()
        val y = this.datScanner.nextDouble()
        val inTheSecondQuarter = x < 0 && y > 0
        return this.ansScanner.checkBoolean(inTheSecondQuarter)
    }

    private fun boolean26(): Boolean {
        val x = this.datScanner.nextDouble()
        val y = this.datScanner.nextDouble()
        val inTheFourthQuarter = x > 0 && y < 0
        return this.ansScanner.checkBoolean(inTheFourthQuarter)
    }

    private fun boolean27(): Boolean {
        val x = this.datScanner.nextDouble()
        val y = this.datScanner.nextDouble()
        val result = x < 0 && y != 0.0
        return this.ansScanner.checkBoolean(result)
    }

    private fun boolean28(): Boolean {
        val x = this.datScanner.nextDouble()
        val y = this.datScanner.nextDouble()
        val result = x * y > 0
        return this.ansScanner.checkBoolean(result)
    }

    private fun boolean29(): Boolean {
        val x = this.datScanner.nextDouble()
        val y = this.datScanner.nextDouble()
        val x1 = this.datScanner.nextDouble()
        val y1 = this.datScanner.nextDouble()
        val x2 = this.datScanner.nextDouble()
        val y2 = this.datScanner.nextDouble()
        val isInside = x > x1 && x < x2 && y > y2 && y < y1
        return this.ansScanner.checkBoolean(isInside)
    }

    private fun boolean30(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val c = this.datScanner.nextInt()
        val isEquilateral = a == b && b == c && c == a
        return this.ansScanner.checkBoolean(isEquilateral)
    }

    private fun boolean31(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val c = this.datScanner.nextInt()
        val isIsosceles = a == b || b == c || c == a
        return this.ansScanner.checkBoolean(isIsosceles)
    }

    private fun boolean32(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val c = this.datScanner.nextInt()
        val isRightTriangle = a*a == b*b + c*c ||
                b*b == a*a + c*c || c*c == a*a + b*b
        return this.ansScanner.checkBoolean(isRightTriangle)
    }

    private fun boolean33(): Boolean {
        val a = this.datScanner.nextInt()
        val b = this.datScanner.nextInt()
        val c = this.datScanner.nextInt()
        val existsTriangle = a + b > c && b + c > a && c + a > b
        return this.ansScanner.checkBoolean(existsTriangle)
    }

    private fun boolean34(): Boolean {
        val x = this.datScanner.nextInt()
        val y = this.datScanner.nextInt()
        val isWhite = (x + y) % 2 != 0
        return this.ansScanner.checkBoolean(isWhite)
    }

    private fun boolean35(): Boolean {
        val x1 = this.datScanner.nextInt()
        val y1 = this.datScanner.nextInt()
        val x2 = this.datScanner.nextInt()
        val y2 = this.datScanner.nextInt()
        val sameColor = (x1 + y1) % 2 == (x2 + y2) % 2
        return this.ansScanner.checkBoolean(sameColor)
    }

    private fun boolean36(): Boolean {
        val x1 = this.datScanner.nextInt()
        val y1 = this.datScanner.nextInt()
        val x2 = this.datScanner.nextInt()
        val y2 = this.datScanner.nextInt()
        val canMoveRook = x1 == x2 || y1 == y2
        return this.ansScanner.checkBoolean(canMoveRook)
    }

    private fun boolean37(): Boolean {
        val x1 = this.datScanner.nextInt()
        val y1 = this.datScanner.nextInt()
        val x2 = this.datScanner.nextInt()
        val y2 = this.datScanner.nextInt()
        val canMoveKing = abs(x2-x1) < 2 && abs(y2-y1) < 2
        return this.ansScanner.checkBoolean(canMoveKing)
    }

    private fun boolean38(): Boolean {
        val x1 = this.datScanner.nextInt()
        val y1 = this.datScanner.nextInt()
        val x2 = this.datScanner.nextInt()
        val y2 = this.datScanner.nextInt()
        val canMoveBishop = abs(x2-x1) == abs(y2-y1)
        return this.ansScanner.checkBoolean(canMoveBishop)
    }

    private fun boolean39(): Boolean {
        val x1 = this.datScanner.nextInt()
        val y1 = this.datScanner.nextInt()
        val x2 = this.datScanner.nextInt()
        val y2 = this.datScanner.nextInt()
        val canMoveQueen = x1 == x2 || y1 == y2 || abs(x2-x1) == abs(y2-y1)
        return this.ansScanner.checkBoolean(canMoveQueen)
    }

    private fun boolean40(): Boolean {
        val x1 = this.datScanner.nextInt()
        val y1 = this.datScanner.nextInt()
        val x2 = this.datScanner.nextInt()
        val y2 = this.datScanner.nextInt()
        val xs = abs(x2 - x1)
        val ys = abs(y2 - y1)
        val canMoveKnight = xs*ys == 2 && xs != ys
        return this.ansScanner.checkBoolean(canMoveKnight)
    }
}

File: go/task/03boolean.go
package task

import (
	"fmt"
	"math"

	"github.com/pkg/errors"
	"github.com/shahzodshafizod/ptaskbook/go/pkg/checker"
	"github.com/shahzodshafizod/ptaskbook/go/pkg/scanner"
)

type booleanTask struct {
	dataPath string
	name     string
	count    int
	scanner  scanner.Scanner
	checker  checker.Checker
}

func NewBooleanTask(pathPrefix string) Task {
	return &booleanTask{
		dataPath: pathPrefix + "/03boolean/Boolean%03d/%03d/%03d",
		name:     "Boolean",
		count:    40,
	}
}

func (bt *booleanTask) GetCount() int { return bt.count }

func (bt *booleanTask) GetName() string { return bt.name }

func (bt *booleanTask) SetData(taskNo int, testNo int) error {
	var filePath = fmt.Sprintf(bt.dataPath, taskNo, testNo, testNo)
	scanner, err := scanner.New(filePath + ".dat")
	if err != nil {
		return errors.Wrap(err, "scanner.New")
	}
	checker, err := checker.New(filePath + ".ans")
	if err != nil {
		return errors.Wrap(err, "checker.New")
	}
	bt.scanner = scanner
	bt.checker = checker
	return nil
}

func (bt *booleanTask) ScannerEOF() bool { return bt.scanner.EOF() }

func (bt *booleanTask) CheckerEOF() bool { return bt.checker.EOF() }

func (bt *booleanTask) CleanData() {
	bt.scanner.RemoveFiles()
	bt.checker.RemoveFiles()
}

func (bt *booleanTask) GetFn(taskNo int) func() bool {
	switch taskNo {
	case 1:
		return bt.boolean1
	case 2:
		return bt.boolean2
	case 3:
		return bt.boolean3
	case 4:
		return bt.boolean4
	case 5:
		return bt.boolean5
	case 6:
		return bt.boolean6
	case 7:
		return bt.boolean7
	case 8:
		return bt.boolean8
	case 9:
		return bt.boolean9
	case 10:
		return bt.boolean10
	case 11:
		return bt.boolean11
	case 12:
		return bt.boolean12
	case 13:
		return bt.boolean13
	case 14:
		return bt.boolean14
	case 15:
		return bt.boolean15
	case 16:
		return bt.boolean16
	case 17:
		return bt.boolean17
	case 18:
		return bt.boolean18
	case 19:
		return bt.boolean19
	case 20:
		return bt.boolean20
	case 21:
		return bt.boolean21
	case 22:
		return bt.boolean22
	case 23:
		return bt.boolean23
	case 24:
		return bt.boolean24
	case 25:
		return bt.boolean25
	case 26:
		return bt.boolean26
	case 27:
		return bt.boolean27
	case 28:
		return bt.boolean28
	case 29:
		return bt.boolean29
	case 30:
		return bt.boolean30
	case 31:
		return bt.boolean31
	case 32:
		return bt.boolean32
	case 33:
		return bt.boolean33
	case 34:
		return bt.boolean34
	case 35:
		return bt.boolean35
	case 36:
		return bt.boolean36
	case 37:
		return bt.boolean37
	case 38:
		return bt.boolean38
	case 39:
		return bt.boolean39
	case 40:
		return bt.boolean40
	default:
		return nil
	}
}

// { "no": 1, "dat": "", "ans": "" }
func (bt *booleanTask) boolean1() bool {
	var a = bt.scanner.NextInt()
	var isPositive = a > 0
	return bt.checker.CompareBool(isPositive)
}

// { "no": 2, "dat": "", "ans": "" }
func (bt *booleanTask) boolean2() bool {
	var a = bt.scanner.NextInt()
	var isOdd = a%2 != 0
	return bt.checker.CompareBool(isOdd)
}

// { "no": 3, "dat": "", "ans": "" }
func (bt *booleanTask) boolean3() bool {
	var a = bt.scanner.NextInt()
	var isEven = a%2 == 0
	return bt.checker.CompareBool(isEven)
}

// { "no": 4, "dat": "", "ans": "" }
func (bt *booleanTask) boolean4() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var result = a > 2 && b <= 3
	return bt.checker.CompareBool(result)
}

// { "no": 5, "dat": "", "ans": "" }
func (bt *booleanTask) boolean5() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var result = a >= 0 || b < -2
	return bt.checker.CompareBool(result)
}

// { "no": 6, "dat": "", "ans": "" }
func (bt *booleanTask) boolean6() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var c = bt.scanner.NextInt()
	var isOrderedAsc = a < b && b < c
	return bt.checker.CompareBool(isOrderedAsc)
}

// { "no": 7, "dat": "", "ans": "" }
func (bt *booleanTask) boolean7() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var c = bt.scanner.NextInt()
	var isOrdered = a < b && b < c || a > b && b > c
	return bt.checker.CompareBool(isOrdered)
}

// { "no": 8, "dat": "", "ans": "" }
func (bt *booleanTask) boolean8() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var eachOdd = (a*b)%2 != 0
	return bt.checker.CompareBool(eachOdd)
}

// { "no": 9, "dat": "", "ans": "" }
func (bt *booleanTask) boolean9() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var anyOdd = a%2 != 0 || b%2 != 0
	return bt.checker.CompareBool(anyOdd)
}

// { "no": 10, "dat": "", "ans": "" }
func (bt *booleanTask) boolean10() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var oneOdd = (a+b)%2 != 0
	return bt.checker.CompareBool(oneOdd)
}

// { "no": 11, "dat": "", "ans": "" }
func (bt *booleanTask) boolean11() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var equalParity = (a+b)%2 == 0
	return bt.checker.CompareBool(equalParity)
}

// { "no": 12, "dat": "", "ans": "" }
func (bt *booleanTask) boolean12() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var c = bt.scanner.NextInt()
	var eachPositive = a > 0 && b > 0 && c > 0
	return bt.checker.CompareBool(eachPositive)
}

// { "no": 13, "dat": "", "ans": "" }
func (bt *booleanTask) boolean13() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var c = bt.scanner.NextInt()
	var anyPositive = a > 0 || b > 0 || c > 0
	return bt.checker.CompareBool(anyPositive)
}

// { "no": 14, "dat": "", "ans": "" }
func (bt *booleanTask) boolean14() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var c = bt.scanner.NextInt()
	var onePositive = (a > 0 && b <= 0 && c <= 0) ||
		(a <= 0 && b > 0 && c <= 0) ||
		(a <= 0 && b <= 0 && c > 0)
	return bt.checker.CompareBool(onePositive)
}

// { "no": 15, "dat": "", "ans": "" }
func (bt *booleanTask) boolean15() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var c = bt.scanner.NextInt()
	var twoPositives = (a > 0 && b > 0 && c <= 0) ||
		(a > 0 && b <= 0 && c > 0) ||
		(a <= 0 && b > 0 && c > 0)
	return bt.checker.CompareBool(twoPositives)
}

// { "no": 16, "dat": "", "ans": "" }
func (bt *booleanTask) boolean16() bool {
	var number = bt.scanner.NextInt()
	var isTwoDigitalEven = number > 9 && number < 100 && number%2 == 0
	return bt.checker.CompareBool(isTwoDigitalEven)
}

// { "no": 17, "dat": "", "ans": "" }
func (bt *booleanTask) boolean17() bool {
	var number = bt.scanner.NextInt()
	var threeDigitalOdd = number > 99 && number < 1000 && number%2 != 0
	return bt.checker.CompareBool(threeDigitalOdd)
}

// { "no": 18, "dat": "", "ans": "" }
func (bt *booleanTask) boolean18() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var c = bt.scanner.NextInt()
	var anyPair = a == b || b == c || c == a
	return bt.checker.CompareBool(anyPair)
}

// { "no": 19, "dat": "", "ans": "" }
func (bt *booleanTask) boolean19() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var c = bt.scanner.NextInt()
	var anyOppositePair = a == -b || b == -c || c == -a
	return bt.checker.CompareBool(anyOppositePair)
}

// { "no": 20, "dat": "", "ans": "" }
func (bt *booleanTask) boolean20() bool {
	var number = bt.scanner.NextInt()
	var hundreds = number / 100
	var tens = number / 10 % 10
	var ones = number % 10
	var areDifferent = hundreds != tens && hundreds != ones && tens != ones
	return bt.checker.CompareBool(areDifferent)
}

// { "no": 21, "dat": "", "ans": "" }
func (bt *booleanTask) boolean21() bool {
	var number = bt.scanner.NextInt()
	var hundreds = number / 100
	var tens = number / 10 % 10
	var ones = number % 10
	var areOrderedAsc = hundreds < tens && tens < ones
	return bt.checker.CompareBool(areOrderedAsc)
}

// { "no": 22, "dat": "", "ans": "" }
func (bt *booleanTask) boolean22() bool {
	var number = bt.scanner.NextInt()
	var hundreds = number / 100
	var tens = number / 10 % 10
	var ones = number % 10
	var areOrdered = hundreds < tens && tens < ones ||
		hundreds > tens && tens > ones
	return bt.checker.CompareBool(areOrdered)
}

// { "no": 23, "dat": "", "ans": "" }
func (bt *booleanTask) boolean23() bool {
	var number = bt.scanner.NextInt()
	var firstHalf = number / 100
	var secondHalf = number%10*10 + number/10%10
	var isPalindrome = firstHalf == secondHalf
	return bt.checker.CompareBool(isPalindrome)
}

// { "no": 24, "dat": "2", "ans": "" }
func (bt *booleanTask) boolean24() bool {
	var a = bt.scanner.NextFloat32()
	var b = bt.scanner.NextFloat32()
	var c = bt.scanner.NextFloat32()
	var d = b*b - 4*a*c
	var hasRoots = d >= 0
	return bt.checker.CompareBool(hasRoots)
}

// { "no": 25, "dat": "2", "ans": "" }
func (bt *booleanTask) boolean25() bool {
	var x = bt.scanner.NextFloat32()
	var y = bt.scanner.NextFloat32()
	var inTheSecondQuarter = x < 0 && y > 0
	return bt.checker.CompareBool(inTheSecondQuarter)
}

// { "no": 26, "dat": "2", "ans": "" }
func (bt *booleanTask) boolean26() bool {
	var x = bt.scanner.NextFloat32()
	var y = bt.scanner.NextFloat32()
	var inTheFourthQuarter = x > 0 && y < 0
	return bt.checker.CompareBool(inTheFourthQuarter)
}

// { "no": 27, "dat": "2", "ans": "" }
func (bt *booleanTask) boolean27() bool {
	var x = bt.scanner.NextFloat32()
	var y = bt.scanner.NextFloat32()
	var result = x < 0 && y != 0
	return bt.checker.CompareBool(result)
}

// { "no": 28, "dat": "2", "ans": "" }
func (bt *booleanTask) boolean28() bool {
	var x = bt.scanner.NextFloat32()
	var y = bt.scanner.NextFloat32()
	var result = x*y > 0
	return bt.checker.CompareBool(result)
}

// { "no": 29, "dat": "2", "ans": "" }
func (bt *booleanTask) boolean29() bool {
	var x = bt.scanner.NextFloat32()
	var y = bt.scanner.NextFloat32()
	var x1 = bt.scanner.NextFloat32()
	var y1 = bt.scanner.NextFloat32()
	var x2 = bt.scanner.NextFloat32()
	var y2 = bt.scanner.NextFloat32()
	var isInside = x > x1 && x < x2 && y > y2 && y < y1
	return bt.checker.CompareBool(isInside)
}

// { "no": 30, "dat": "", "ans": "" }
func (bt *booleanTask) boolean30() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var c = bt.scanner.NextInt()
	var isEquilAteral = a == b && b == c
	return bt.checker.CompareBool(isEquilAteral)
}

// { "no": 31, "dat": "", "ans": "" }
func (bt *booleanTask) boolean31() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var c = bt.scanner.NextInt()
	var isIsosceles = a == b || b == c || c == a
	return bt.checker.CompareBool(isIsosceles)
}

// { "no": 32, "dat": "", "ans": "" }
func (bt *booleanTask) boolean32() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var c = bt.scanner.NextInt()
	var isRightTriangle = a*a == b*b+c*c ||
		b*b == a*a+c*c || c*c == a*a+b*b
	return bt.checker.CompareBool(isRightTriangle)
}

// { "no": 33, "dat": "", "ans": "" }
func (bt *booleanTask) boolean33() bool {
	var a = bt.scanner.NextInt()
	var b = bt.scanner.NextInt()
	var c = bt.scanner.NextInt()
	var existsTriangle = a+b > c && a+c > b && b+c > a
	return bt.checker.CompareBool(existsTriangle)
}

// { "no": 34, "dat": "", "ans": "" }
func (bt *booleanTask) boolean34() bool {
	var x = bt.scanner.NextInt()
	var y = bt.scanner.NextInt()
	var isWhiteSquare = (x+y)%2 != 0
	return bt.checker.CompareBool(isWhiteSquare)
}

// { "no": 35, "dat": "", "ans": "" }
func (bt *booleanTask) boolean35() bool {
	var x1 = bt.scanner.NextInt()
	var y1 = bt.scanner.NextInt()
	var x2 = bt.scanner.NextInt()
	var y2 = bt.scanner.NextInt()
	var sameColor = (x1+y1)%2 == (x2+y2)%2
	return bt.checker.CompareBool(sameColor)
}

// { "no": 36, "dat": "", "ans": "" }
func (bt *booleanTask) boolean36() bool {
	var x1 = bt.scanner.NextInt()
	var y1 = bt.scanner.NextInt()
	var x2 = bt.scanner.NextInt()
	var y2 = bt.scanner.NextInt()
	var canMoveRook = x1 == x2 || y1 == y2
	return bt.checker.CompareBool(canMoveRook)
}

// { "no": 37, "dat": "", "ans": "" }
func (bt *booleanTask) boolean37() bool {
	var x1 = bt.scanner.NextInt()
	var y1 = bt.scanner.NextInt()
	var x2 = bt.scanner.NextInt()
	var y2 = bt.scanner.NextInt()
	var canMoveKing = math.Abs(float64(x2-x1)) < 2 && math.Abs(float64(y2-y1)) < 2
	return bt.checker.CompareBool(canMoveKing)
}

// { "no": 38, "dat": "", "ans": "" }
func (bt *booleanTask) boolean38() bool {
	var x1 = bt.scanner.NextInt()
	var y1 = bt.scanner.NextInt()
	var x2 = bt.scanner.NextInt()
	var y2 = bt.scanner.NextInt()
	var canMoveBishop = math.Abs(float64(x2-x1)) == math.Abs(float64(y2-y1))
	return bt.checker.CompareBool(canMoveBishop)
}

// { "no": 39, "dat": "", "ans": "" }
func (bt *booleanTask) boolean39() bool {
	var x1 = bt.scanner.NextInt()
	var y1 = bt.scanner.NextInt()
	var x2 = bt.scanner.NextInt()
	var y2 = bt.scanner.NextInt()
	var canMoveQueen = x1 == x2 || y1 == y2 ||
		math.Abs(float64(x2-x1)) == math.Abs(float64(y2-y1))
	return bt.checker.CompareBool(canMoveQueen)
}

// { "no": 40, "dat": "", "ans": "" }
func (bt *booleanTask) boolean40() bool {
	var x1 = bt.scanner.NextInt()
	var y1 = bt.scanner.NextInt()
	var x2 = bt.scanner.NextInt()
	var y2 = bt.scanner.NextInt()
	var xs = math.Abs(float64(x2 - x1))
	var ys = math.Abs(float64(y2 - y1))
	var canMoveKnight = xs*ys == 2 && xs != ys
	return bt.checker.CompareBool(canMoveKnight)
}
